File: week01/twoSum.js
var twoSum = function (nums, target) {
	let seen = {};
	for (let i = 0; i < nums.length; i++) {
		let remaining = target - nums[i];
		if (seen.hasOwnProperty(remaining)) {
			return [seen[remaining], i];
		}
		seen[nums[i]] = i;
	}
	return [];
};
console.log(twoSum([2, 7, 11, 9], 9));

File: week01/two_sum.cpp
// TC: O(N)
// MC: O(N)
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>seen;
        for(int i = 0; i < nums.size(); i++)
        {
            int v = target - nums[i];
            if(seen.find(v) != seen.end())
            {
               return {seen[v], i}; 
            }
            seen[nums[i]] = i;
        }
        return {};
    }
};